src/protobuf-net.Reflection/CSharpCodeGenerator.cs

```diff
@@ -709,6 +709,36 @@ protected override void WriteExtension(GeneratorContext ctx, FieldDescriptorProt
             }
         }
 
+        /// <summary>
+        /// Starts a service block
+        /// </summary>
+        protected override void WriteServiceHeader(GeneratorContext ctx, ServiceDescriptorProto service, ref object state)
+        {
+            var name = service.Name;
+            ctx.WriteLine($"{GetAccess(GetAccess(service))} interface I{Escape(name)}").WriteLine("{").Indent();
+        }
+        /// <summary>
+        /// Ends an service block
+        /// </summary>
+        protected override void WriteServiceFooter(GeneratorContext ctx, ServiceDescriptorProto service, ref object state)
+        {
+            ctx.Outdent().WriteLine("}").WriteLine();
+        }
+        /// <summary>
+        /// Write a service method
+        /// </summary>
+        protected override void WriteServiceMethod(GeneratorContext ctx, MethodDescriptorProto method, ref object state)
+        {
+            var outputType = method.OutputType;
+            var inputType = method.InputType;
+
+            // TODO: What does this signify?
```

Comment:
that's just how types are fully qualified when resolved via protobuf (especially when it comes from external schemas) you might want to look at `GetTypeName`, noting that IIRC we're always talking about a *message* type here; it may be sufficient to either move or copy the code from `case FieldDescriptorProto.Type.TypeMessage`, so you handle the special types, but the key parts are probably `ctx.TryFind<DescriptorProto>(name)` and `MakeRelativeName`

Reply:
Are you saying `ctx.TryFind<DescriptorProto>(method.OutputType)` is what should be done?

`MakeRelativeName` only accepts `FieldDescriptorProto`, not sure if its better to make a new method, or just copy the code over, because it will need to return both input and output names.